example/lib/example.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Example do
+  @on_load :__on_load__
+
+  def __on_load__ do
+    path = :filename.join(:code.priv_dir(:example), ~c"libexample")
```

Comment:
this can also be the following, wdyt?

```suggestion
    path = Application.app_dir(:example, ["priv", "libexample"])
```

Reply:
Looks good, but according to [`erlang:load_nif/2`](https://www.erlang.org/doc/apps/erts/erlang#load_nif/2) typespec the path should be a charlist, so probably safer to stick to that.